### DISCUSSION TITLE
A temporary alternative solution for web worker version @hpcc-js/wasm/graphicviz

### DISCUSSION THREAD

**Discussion by coder-xiaotian** [hpcc-systems/hpcc-js-wasm#159] [Show and tell]
https://github.com/coder-xiaotian/graphviz-worker

**Comment by GordonSmith** [comment]
I never really considered having built in worker support, I just had tests to make sure this lib worked well as a worker - here are the tests I use for ensuring this lib plays nice in workers:
* https://github.com/hpcc-systems/hpcc-js-wasm/blob/trunk/src-ts/__tests__/worker-browser.ts
* https://github.com/hpcc-systems/hpcc-js-wasm/blob/trunk/src-ts/__tests__/worker-node.ts

(you might want to tweak your code, so that so it also works in NodeJS) 

Would you like to issue a PR to integrate here, or was your post just an FYI?

Thanks for the example though!

**Comment by coder-xiaotian** [comment]
> I never really considered having built in worker support, I just had tests to make sure this lib worked well as a worker - here are the tests I use for ensuring this lib plays nice in workers:
> 
> * https://github.com/hpcc-systems/hpcc-js-wasm/blob/trunk/src-ts/__tests__/worker-browser.ts
> * https://github.com/hpcc-systems/hpcc-js-wasm/blob/trunk/src-ts/__tests__/worker-node.ts
> 
> (you might want to tweak your code, so that so it also works in NodeJS)
> 
> Would you like to issue a PR to integrate here, or was your post just an FYI?
> 
> Thanks for the example though!

I want to use @hpcc-js/wasm as a web worker in my project, so what do you mean is that I can issue a PR to enable @hpcc-js/wasm to support web workers?

**Comment by GordonSmith** [comment]
I wasn't sure if (by opening this issue) you wanted to see built in worker support here?

Something like:  `import { Graphviz } from "@hpcc-js/wasm/graphviz/worker"` with a compatible API, so the user doesn't need to do anything except add the "/worker" postfix?

**Comment by coder-xiaotian** [comment]
> I wasn't sure if (by opening this issue) you wanted to see built in worker support here?
> 
> Something like: `import { Graphviz } from "@hpcc-js/wasm/graphviz/worker"` with a compatible API, so the user doesn't need to do anything except add the "/worker" postfix?

I opened this issue to provide a convenient way for others to use @hpcc-js/wasm/graphicviz in web worker. You can label it FYI and then close it.
